packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -104,22 +105,193 @@ export default class VTable {
             return;
         }
         this.table.style.borderCollapse = 'collapse';
-        this.trs[0].style.backgroundColor = format.bgColorOdd || 'transparent';
-        if (this.trs[1]) {
-            this.trs[1].style.backgroundColor = format.bgColorEven || 'transparent';
-        }
+        this.setBorderColors(format);
+        this.setTableRowColor(format);
+        this.setTableColumnsColor(format);
+        this.setBorderType(format);
+        this.setFirstColumnFormat(format);
+        this.setHeaderRowFormat(format);
+    }
+
+    /**
+     * Set color to borders of an table
+     * @param format
+     * @returns
+     */
+    private setBorderColors(format: Partial<TableFormat>) {
         this.cells.forEach(row =>
             row
                 .filter(cell => cell.td)
                 .forEach(cell => {
                     cell.td.style.borderTop = getBorderStyle(format.topBorderColor);
-                    cell.td.style.borderBottom = getBorderStyle(format.bottomBorderColor);
                     cell.td.style.borderLeft = getBorderStyle(format.verticalBorderColor);
                     cell.td.style.borderRight = getBorderStyle(format.verticalBorderColor);
+                    cell.td.style.borderBottom = getBorderStyle(format.bottomBorderColor);
+                    cell.td.style.backgroundColor = format.bgColor;
                 })
         );
     }
 
+    /**
+     * Set color to even and odd rows
+     * @param format
+     */
+    private setTableRowColor(format: Partial<TableFormat>) {
+        if (!format.bandedRows) {
+            return;
+        }
+        const evenRows = this.cells.filter(row => this.cells.indexOf(row) % 2 === 0);
+        const oddRows = this.cells.filter(row => this.cells.indexOf(row) % 2 === 1);
+        evenRows.forEach(row => {
+            row.filter(cell => cell.td).forEach(
+                cell => (cell.td.style.backgroundColor = format.bgColorEven)
+            );
+        });
+        oddRows.forEach(row => {
+            row.filter(cell => cell.td).forEach(
+                cell => (cell.td.style.backgroundColor = format.bgColorOdd)
+            );
+        });
+    }

```

Comment:
nit: You could do something like this so only traverse the cells once instead of 3 times.

![image](https://user-images.githubusercontent.com/8291124/147291340-984682e9-e791-4e5a-96a1-17469cc97470.png)